### PR TITLE
C# IR: Some minor additions

### PR DESCRIPTION
**Pull request by AndreiDiaconu1**
This is a small PR whose aim is to tie some loose ends.
The first commit removes a class that was useless (`TranslatedLambdaExpr`).
The second commit adds a test for indexers (no new code was needed because of the AST desugaring) and adds support for assignment operations involving events.

### CODE REVIEW

File: csharp/ql/test/library-tests/ir/ir/events.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Events
+{
+    public delegate string MyDel(string str);
+
+    event MyDel MyEvent;
+
+    public Events() {
+        this.MyEvent += new MyDel(this.WelcomeUser);
```
Comment (hvitved) [inline csharp/ql/test/library-tests/ir/ir/events.cs]: Add a test for `-=` as well?

File: csharp/ql/test/library-tests/ir/ir/indexers.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            }
+        }
+    }
+
+    public static void Main()
+    {
+        Contact contact = new Contact();
+        contact[0] = "Begumpet";
+        contact[1] = "Hyderabad";
+        contact[2] = "Telengana";
```
Comment (hvitved) [inline csharp/ql/test/library-tests/ir/ir/indexers.cs]: add a test for `get_Item` as well? E.g. `contact[3] = contact[2]`.

Comment (AndreiDiaconu1) [reply csharp/ql/test/library-tests/ir/ir/indexers.cs]: Added and modified test a bit. Also discovered a spurious `Load` when testing `get_Item`.

File: csharp/ql/test/library-tests/ir/ir/events.cs (hunk 1 above)
Comment (AndreiDiaconu1) [reply csharp/ql/test/library-tests/ir/ir/events.cs]: Added and modified the test a bit.

**Pull request merged**